magpie/api/webhooks.py

```diff
@@ -63,16 +92,20 @@ def send_webhook_request(webhook_config, params, update_user_status_on_error=Fal
     :param update_user_status_on_error: update the user status or not in case of a webhook error
     """
     # Replace each instance of template parameters if a corresponding value was defined in input
-    for template_param in WEBHOOK_TEMPLATE_PARAMS:
-        if template_param in params:
-            for k, v in webhook_config["payload"].items():
-                webhook_config["payload"][k] = v.replace("{" + template_param + "}", params[template_param])
-
     try:
-        resp = requests.request(webhook_config["method"], webhook_config["url"], data=webhook_config["payload"])
+        for template_param in WEBHOOK_TEMPLATE_PARAMS:
+            if template_param in params:
+                webhook_config["payload"] = replace_template(template_param,
+                                                             params[template_param],
+                                                             webhook_config["payload"])
+    except Exception as exception:
+        LOGGER.error("An exception has occured while processing the template parameters in a webhook payload : %s",
+                     str(exception))
+    try:
+        resp = requests.request(webhook_config["method"], webhook_config["url"], json=webhook_config["payload"])
```

Comment:
I was planning the rest API of a webhook and I though it would be nice to make a replace_template in the url too!
`POST /users {'user': <user_name>}`
`DELETE /users/<user_name>`